tools/scripts/ffi.toml

```diff
@@ -312,4 +313,137 @@ else
     echo "Installing Diplomat rev ${rev}"
     exec cargo install --git https://github.com/rust-diplomat/diplomat.git --rev ${rev} diplomat-tool -f
 end
-'''
\ No newline at end of file
+'''
+
+[tasks.gn-install]
+description = "Install tools required for GN locally in the icu4x project"
+category = "ICU4X Development"
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+
+mkdir ffi/gn/third_party_tools
+cd ffi/gn/third_party_tools
+
+# depot_tools contains the following things we need:
+# 1. A cross-platform ninja wrapper script
+# 2. cipd, a tool to download a prebuilt GN binary
+if is_path_exists depot_tools
+    echo "NOTE: depot_tools already exists"
+else
+    exec --fail-on-error git clone --depth 1 https://chromium.googlesource.com/chromium/tools/depot_tools.git
+end
+
+# Write an "ensure file", which is used as input to cipd
+writefile ensure_file "${ICU4X_GN_PACKAGE} latest"
+exec --fail-on-error depot_tools/cipd ensure -root bin -ensure-file ensure_file
+
+# To get GNaw, we need to download Fuchsia sources
+if is_path_exists fuchsia
+    echo "NOTE: fuchsia already exists"
+else
+    exec --fail-on-error git clone --depth 1 https://fuchsia.googlesource.com/fuchsia
+    exec ln -s Cargo.toml.crates-io fuchsia/tools/cargo-gnaw/Cargo.toml
+end
+
+# Install GNaw locally in the bin directory (alongside GN)
+# Since the parent directory is vendored, we have no choice but to vendor here, too
+mkdir .cargo
+writefile .cargo/config "source = { crates-io.replace-with = 'vendored-sources', vendored-sources.directory = 'vendor' }"
+exec cargo vendor --manifest-path fuchsia/tools/cargo-gnaw/Cargo.toml
```

Comment:
question: why do we need to vendor this?

Reply:
if the answer is basically "gnaw requires depot tools set up correctly to work" we should mention that in a comment.

Reply:
The comment on line 350 (3 lines above) reads

```
# Since the parent directory is vendored, we have no choice but to vendor here, too
```